src/main/java/uk/ac/ebi/biostudies/service/TextMiningLinkUpdater.java

```diff
@@ -0,0 +1,142 @@
+            if (curNode == null)
+                continue;
+            allNodes.add(curNode);
+        }
+        for(JsonNode curNode: allNodes){
+            IndexableField field = document.getField(curNode.get(Constants.IndexEntryAttributes.NAME).asText());
+            String fieldType = curNode.get(Constants.IndexEntryAttributes.FIELD_TYPE).asText();
+            try {
+                switch (fieldType) {
+                    case Constants.IndexEntryAttributes.FieldTypeValues.UNTOKENIZED_STRING:
+                        if (curNode.has(Constants.IndexEntryAttributes.SORTABLE) && curNode.get(Constants.IndexEntryAttributes.SORTABLE).asBoolean(false))
+                            document.add(new SortedDocValuesField(String.valueOf(field.name()), new BytesRef(field.stringValue())));
+                        break;
+                    case Constants.IndexEntryAttributes.FieldTypeValues.LONG:
+                        Long value = Long.valueOf(field.stringValue());
+                        document.add(new SortedNumericDocValuesField(field.name(), value));
+                        document.add(new LongPoint(field.name(),  value));
+                        break;
+                }
+            } catch (Exception ex) {
+                LOGGER.error("field name: {} doc accession: {}", field.name(), document.get(Constants.Fields.ACCESSION), ex);
+            }
+        }
+        for(LinkInfo linkInfo: allLinks){
+            document.add(new TextField(Constants.Fields.CONTENT, linkInfo.getValue().toLowerCase(), Field.Store.NO));
+            document.add(new TextField(Constants.Fields.CONTENT, linkInfo.getLink(), Field.Store.NO));
+            document.add(new TextField(Constants.Fields.CONTENT, linkInfo.getType(), Field.Store.NO));
+        }
+
+        indexManager.getIndexWriter().updateDocument(new Term(Constants.Fields.ID, document.get(Constants.Fields.ID)), document);
+        indexManager.getIndexWriter().commit();
+
+    }
+
+    public void addToLinkIndex(String accession, String fileName, List<LinkInfo> allLinks) throws Exception{
+        int counter = 0;
+        if(MESSAGE_DIGEST==null)
+            MESSAGE_DIGEST = MessageDigest.getInstance("MD5");
+        byte[] digestKey = MESSAGE_DIGEST.digest((accession+fileName).getBytes("UTF-8"));
+        String digestKeyStr = new String(digestKey, "UTF-8");
+        TermQuery digestTermQuery = new TermQuery(new Term(Constants.File.DIGEST, digestKeyStr));
+        long delNumber = indexManager.getLinkIndexWriter().deleteDocuments(digestTermQuery);
+        indexManager.getLinkIndexWriter().commit();
+        LOGGER.debug("{} old links are deleted", delNumber);
+        for(LinkInfo linkInfo:allLinks){
+            counter++;
+            Document document = new Document();
+            document.add(new StringField(Constants.Fields.ID, accession+"_"+counter, Field.Store.YES));
+            document.add(new StringField(Constants.File.DIGEST, digestKeyStr, Field.Store.YES));
```

Comment:
Not quite sure what digest means here. Can the name be more descriptive?

Reply:
digestKey is generating a searchable unique String key based on accession and filename
is DigestUniqueKey a better choice?
No idea

Reply:
May be `Constants.Link.KEY ="key"` ?